src/wallet/wallet2.cpp

```diff
@@ -13007,6 +13010,79 @@ void wallet2::throw_on_rpc_response_error(const boost::optional<std::string> &st
   THROW_WALLET_EXCEPTION_IF(*status != CORE_RPC_STATUS_OK, tools::error::wallet_generic_rpc_error, method, m_trusted_daemon ? *status : "daemon error");
 }
 //----------------------------------------------------------------------------------------------------
+
+bool wallet2::save_to_file(const std::string& path_to_file, const std::string& raw) const
+{
+  if (m_export_format == ExportFormat::Hex)
+  {
+    std::string h = HEX_OUTPUT_MAGIC + "\n";
+    h.append(epee::string_tools::buff_to_hex_nodelimer(raw));
+
+    // Split the long string into multiple lines, similarly to how openssl
+    // prints certificates.
+    for (size_t i = 63; i < h.length(); i += 64)
+    {
+      h.insert(i, 1, '\n');
+    }
+
+    h.insert(h.end(), '\n');
+
+    return epee::file_io_utils::save_string_to_file(path_to_file, h);
+  }
+  else
+  {
+    return epee::file_io_utils::save_string_to_file(path_to_file, raw);
+  }
+}
+//----------------------------------------------------------------------------------------------------
+
+bool wallet2::load_from_file(const std::string& path_to_file, std::string& target_str,
+                             size_t max_size) const
+{
+  std::string data;
+  bool r = epee::file_io_utils::load_file_to_string(path_to_file, data, max_size);
+  if (!r)
+  {
+    return false;
+  }
+
+  // Ignore any whitespace.
+  // We try to convert the version without whitespace from hex to binary.
+  // If that fails, we stick to the original data (hence the copy).
+  std::string without_whitespace(data);
+  without_whitespace.erase(
+    std::remove_if(
+      without_whitespace.begin(),
+      without_whitespace.end(),
+      [](char c){ return (c =='\r' || c =='\t' || c == ' ' || c == '\n'); }),
+    without_whitespace.end());
+
+  if (0 != without_whitespace.compare(0, HEX_OUTPUT_MAGIC.length(), HEX_OUTPUT_MAGIC))
+  {
+    // It's NOT our hex dump.
+    target_str = std::move(data);
+    return true;
+  }
+  // It's apparently our hex dump.
+
+  without_whitespace.erase(0, HEX_OUTPUT_MAGIC.length());
+  r = epee::string_tools::validate_hex(without_whitespace.size(), without_whitespace);
+  if (!r)
+  {
+    return false;
+  }
```

Comment:
The magic is written with an extra \n. This code does not remove the \n, nor check it. It seems to be implicitely skipped by the whitespace cleanup above. Is this intended ?

Reply:
Yes, it's intentional. At this point, the `\n` should have been removed by the whitespace cleanup above. 

The `\n` is there just to make it look nicer when pasted somewhere. I was even contemplating adding the line of dashes like with openssl certificates but it seemed excessive.